### PR TITLE
Make smart follow functions easier to follow

### PR DESCRIPTION
**Pull request by Broxzier**
- Opted for a switch-case instead of if-else if-else if-else
- Reuse entity - is already checked for null and doesn't change
- Remove unnecessary else blocks after if-block that returns
- Make `peep` argument const - following should not change the peep in any way

Turn whitespace changes off in the diff, since most of it are just indention changes.

If you want to test this, you can do so by using the "follow sprite" option in the sequence editor, then playing the sequence.

### CODE REVIEW

File: src/openrct2/interface/Viewport.cpp [hunk 1]
```diff
@@ -719,80 +713,74 @@ viewport_focus viewport_update_smart_guest_follow(rct_window* window, Peep* peep
...
+            focus.coordinate.y = xy.y;
+            focus.coordinate.z = tile_element_height(xy) + (4 * COORDS_Z_STEP);
+            focus.sprite.type |= VIEWPORT_FOCUS_TYPE_COORDINATE;
         }
-        focus.coordinate.rotation = get_current_rotation();
     }
+    else
+    {
+        focus.sprite.type |= VIEWPORT_FOCUS_TYPE_SPRITE | VIEWPORT_FOCUS_TYPE_COORDINATE;
+        focus.sprite.pad_486 &= 0xFFFF;
```
Comment (Broxzier) [inline src/openrct2/interface/Viewport.cpp:753]: Seems to be unused

File: src/openrct2/interface/Viewport.cpp [hunk 2]
```diff
@@ -669,44 +669,38 @@ void viewport_update_sprite_follow(rct_window* window)
...
         {
-            // will never happen
-            window->viewport_smart_follow_sprite = SPRITE_INDEX_NULL;
-            window->viewport_target_sprite = SPRITE_INDEX_NULL;
-            return;
+            Peep* peep = static_cast<Peep*>(entity);
+            if (peep->Is<Guest>())
+                viewport_update_smart_guest_follow(window, peep);
+            else if (peep->Is<Staff>())
+                viewport_update_smart_staff_follow(window, peep);
```
Comment (Broxzier) [inline src/openrct2/interface/Viewport.cpp:692]: Can probably be split, no local Peep needed by passing `entity->As<Guest/Staff>()` to the functions.

File: src/openrct2/interface/Viewport.cpp [hunk 3]
```diff
@@ -669,44 +669,38 @@ void viewport_update_sprite_follow(rct_window* window)
...
+            break;
+
+        case EntityType::Guest:
+        case EntityType::Staff:
         {
-            // will never happen
-            window->viewport_smart_follow_sprite = SPRITE_INDEX_NULL;
-            window->viewport_target_sprite = SPRITE_INDEX_NULL;
-            return;
+            Peep* peep = static_cast<Peep*>(entity);
```
Comment (ZehMatt) [inline src/openrct2/interface/Viewport.cpp:688]:
```suggestion
            if (auto* peep = entity->As<Guest>())
                viewport_update_smart_guest_follow(window, peep);
            else if (auto* staff = entity->As<Staff>())
                viewport_update_smart_staff_follow(window, staff);
```
Could also do it this way

File: src/openrct2/interface/Viewport.h [hunk 4]
```diff
@@ -111,8 +111,8 @@ void viewports_invalidate(int32_t left, int32_t top, int32_t right, int32_t bott
 void viewport_update_position(rct_window* window);
 void viewport_update_sprite_follow(rct_window* window);
 void viewport_update_smart_sprite_follow(rct_window* window);
-viewport_focus viewport_update_smart_guest_follow(rct_window* window, Peep* peep);
-void viewport_update_smart_staff_follow(rct_window* window, Peep* peep);
+viewport_focus viewport_update_smart_guest_follow(rct_window* window, const Peep* peep);
+void viewport_update_smart_staff_follow(rct_window* window, const Peep* peep);
```
Comment (ZehMatt) [inline src/openrct2/interface/Viewport.h:115]:
```suggestion
void viewport_update_smart_staff_follow(rct_window* window, const Staff* staff);
```
Makes a bit more sense I think

File: src/openrct2/interface/Viewport.cpp (hunk 3 above)
Comment (Broxzier) [reply src/openrct2/interface/Viewport.cpp:688]: Splitting them into their own `EntityType::Guest` and `EntityType::Staff` case, the check for `Is<Guest/Staff>` won't even be necessary anymore.

File: src/openrct2/interface/Viewport.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/openrct2/interface/Viewport.cpp:753]: It's very very easy to mess this up the union used for this has a ton of gotchas.

Comment (Broxzier) [reply src/openrct2/interface/Viewport.cpp:750]: Hmm, I've noticed no differences with this line removed, also haven't seen any change to `focus.location` which shares its memory.

**Review by tupaschoal — APPROVED** [review]
Seems alright

**Pull request merged**